### PR TITLE
Getting CI to pass again

### PR DESCRIPTION
**Pull request by KyleLeneau**

### CODE REVIEW

**Comment by KyleLeneau** [comment]
cc @Bouke I took the liberty to update your CI to `xcode9.4` and swift 4.1. All the jobs are passing now including the linux builds. I had to tweak some code and swiftlint rules but nothing major.

File: .swiftlint.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ disabled_rules:
   - function_body_length
   - private_over_fileprivate
   - todo
+  - identifier_name
```
Comment (Bouke) [inline .swiftlint.yml]: Can you revert this change, or why is it needed?

**Comment by Bouke** [comment]
Thanks. Mostly LGTM, will merge after resolving that item.

**Comment by KyleLeneau** [comment]
@Bouke take a look, everything is passing now.

**Comment by Bouke** [comment]
Thank you!
